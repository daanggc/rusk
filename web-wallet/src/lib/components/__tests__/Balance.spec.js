import { afterEach, describe, expect, it } from "vitest";
import { cleanup, render } from "@testing-library/svelte";
import { skipIn } from "lamb";
import { Balance } from "..";

describe("Balance", () => {
  const baseProps = {
    fiatCurrency: "USD",
    fiatPrice: 10,
    locale: "en",
    tokenCurrency: "DUSK",
    tokens: 2000000,
  };
  const baseOptions = {
    props: baseProps,
    target: document.body,
  };

  afterEach(cleanup);

  it("renders the Balance component", () => {
    const { container } = render(Balance, baseOptions);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should update the Balance component when the props change", async () => {
    const { container, rerender } = render(Balance, baseOptions);

    expect(container.firstChild).toMatchSnapshot();

    await rerender({
      fiatCurrency: "EUR",
      fiatPrice: 20,
      locale: "it",
      tokenCurrency: "DUSK",
      tokens: 4000000,
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should pass additional class names and attributes to the rendered element", async () => {
    const props = {
      ...baseProps,
      className: "foo bar",
      id: "balance",
    };
    const { container, rerender } = render(Balance, { ...baseOptions, props });

    expect(container.firstChild).toHaveClass("foo bar");
    expect(container.firstChild).toHaveAttribute("id", "balance");

    await rerender({
      ...props,
      className: "qux",
      id: "new-balance",
    });

    expect(container.firstChild).toHaveClass("qux");
    expect(container.firstChild).toHaveAttribute("id", "new-balance");
  });

  it("should not display the fiat value if the fiat price is `undefined`", () => {
    const props = skipIn(baseProps, ["fiatPrice"]);
    const { container } = render(Balance, { ...baseOptions, props });

    expect(container.querySelector(".dusk-balance__fiat--visible")).toBeNull();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should not display the fiat value if there are no tokens", () => {
    const props = { ...baseProps, tokens: 0 };
    const { container } = render(Balance, { ...baseOptions, props });

    expect(container.querySelector(".dusk-balance__fiat--visible")).toBeNull();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should display the usage indicator if there are shielded tokens", () => {
    const props = { ...baseProps, shieldedTokensPercentage: 50, tokens: 100 };
    const { container } = render(Balance, { ...baseOptions, props });

    expect(container.querySelector(".dusk-balance__usage")).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should not display the usage indicator if there are no shielded tokens", () => {
    const props = { ...baseProps, tokens: 100 };
    const { container } = render(Balance, { ...baseOptions, props });

    expect(
      container.querySelector(".dusk-balance__usage")
    ).not.toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
