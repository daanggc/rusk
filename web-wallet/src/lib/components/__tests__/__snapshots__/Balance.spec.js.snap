// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Balance > renders the Balance component 1`] = `
<article
  class="dusk-balance"
>
  <header
    class="dusk-balance__header"
  >
    <h2
      class="sr-only"
    >
      Your Balance:
    </h2>
  </header>
   
  <p
    class="dusk-balance__dusk"
  >
    <strong>
      2,000,000.000000000
    </strong>
     
    <strong>
      DUSK
    </strong>
  </p>
   
  <p
    class="dusk-balance__fiat dusk-balance__fiat--visible"
  >
    <strong>
      $20,000,000.00
    </strong>
  </p>
   
</article>
`;

exports[`Balance > should display the usage indicator if there are shielded tokens 1`] = `
<article
  class="dusk-balance"
>
  <header
    class="dusk-balance__header"
  >
    <h2
      class="sr-only"
    >
      Your Balance:
    </h2>
  </header>
   
  <p
    class="dusk-balance__dusk"
  >
    <strong>
      100.000000000
    </strong>
     
    <strong>
      DUSK
    </strong>
  </p>
   
  <p
    class="dusk-balance__fiat dusk-balance__fiat--visible"
  >
    <strong>
      $1,000.00
    </strong>
  </p>
   
  <div
    class="usage-indicator dusk-balance__usage"
  >
    <svg
      class="dusk-icon dusk-icon--size--default usage-indicator__icon"
      data-tooltip-id="main-tooltip"
      data-tooltip-text="You have put 50% of your funds in your shielded account"
      role="graphics-symbol"
      viewBox="0 0 24 24"
    >
      <path
        d="M12,1L3,5V11C3,16.55 6.84,21.74 12,23C17.16,21.74 21,16.55 21,11V5L12,1M12,7C13.4,7 14.8,8.1 14.8,9.5V11C15.4,11 16,11.6 16,12.3V15.8C16,16.4 15.4,17 14.7,17H9.2C8.6,17 8,16.4 8,15.7V12.2C8,11.6 8.6,11 9.2,11V9.5C9.2,8.1 10.6,7 12,7M12,8.2C11.2,8.2 10.5,8.7 10.5,9.5V11H13.5V9.5C13.5,8.7 12.8,8.2 12,8.2Z"
      />
    </svg>
    
     
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="50"
      aria-valuetext="You have put 50% of your funds in your shielded account"
      class="usage-indicator__meter"
      role="meter"
    >
      <div
        aria-hidden="true"
        class="usage-indicator__meter-bar"
        style="width: 50%;"
      />
    </div>
     
    <svg
      class="dusk-icon dusk-icon--size--default usage-indicator__icon"
      data-tooltip-id="main-tooltip"
      data-tooltip-text="You have put 50% of your funds in your unshielded account"
      role="graphics-symbol"
      viewBox="0 0 24 24"
    >
      <path
        d="M21 11C21 16.5 17.2 21.7 12 23C6.8 21.7 3 16.5 3 11V5L12 1L21 5V11M12 21C15.8 20 19 15.5 19 11.2V6.3L12 3.2L5 6.3V11.2C5 15.5 8.2 20 12 21M14.8 11H10.5V8.5C10.5 7.7 11.2 7.2 12 7.2S13.5 7.7 13.5 8.5V9H14.8V8.5C14.8 7.1 13.4 6 12 6S9.2 7.1 9.2 8.5V11C8.6 11 8 11.6 8 12.2V15.7C8 16.4 8.6 17 9.2 17H14.7C15.4 17 16 16.4 16 15.8V12.3C16 11.6 15.4 11 14.8 11Z"
      />
    </svg>
    
  </div>
</article>
`;

exports[`Balance > should not display the fiat value if the fiat price is \`undefined\` 1`] = `
<article
  class="dusk-balance"
>
  <header
    class="dusk-balance__header"
  >
    <h2
      class="sr-only"
    >
      Your Balance:
    </h2>
  </header>
   
  <p
    class="dusk-balance__dusk"
  >
    <strong>
      2,000,000.000000000
    </strong>
     
    <strong>
      DUSK
    </strong>
  </p>
   
  <p
    class="dusk-balance__fiat"
  >
    <strong>
      $0.00
    </strong>
  </p>
   
</article>
`;

exports[`Balance > should not display the fiat value if there are no tokens 1`] = `
<article
  class="dusk-balance"
>
  <header
    class="dusk-balance__header"
  >
    <h2
      class="sr-only"
    >
      Your Balance:
    </h2>
  </header>
   
  <p
    class="dusk-balance__dusk"
  >
    <strong>
      0.000000000
    </strong>
     
    <strong>
      DUSK
    </strong>
  </p>
   
  <p
    class="dusk-balance__fiat"
  >
    <strong>
      $0.00
    </strong>
  </p>
   
</article>
`;

exports[`Balance > should not display the usage indicator if there are no shielded tokens 1`] = `
<article
  class="dusk-balance"
>
  <header
    class="dusk-balance__header"
  >
    <h2
      class="sr-only"
    >
      Your Balance:
    </h2>
  </header>
   
  <p
    class="dusk-balance__dusk"
  >
    <strong>
      100.000000000
    </strong>
     
    <strong>
      DUSK
    </strong>
  </p>
   
  <p
    class="dusk-balance__fiat dusk-balance__fiat--visible"
  >
    <strong>
      $1,000.00
    </strong>
  </p>
   
</article>
`;

exports[`Balance > should update the Balance component when the props change 1`] = `
<article
  class="dusk-balance"
>
  <header
    class="dusk-balance__header"
  >
    <h2
      class="sr-only"
    >
      Your Balance:
    </h2>
  </header>
   
  <p
    class="dusk-balance__dusk"
  >
    <strong>
      2,000,000.000000000
    </strong>
     
    <strong>
      DUSK
    </strong>
  </p>
   
  <p
    class="dusk-balance__fiat dusk-balance__fiat--visible"
  >
    <strong>
      $20,000,000.00
    </strong>
  </p>
   
</article>
`;

exports[`Balance > should update the Balance component when the props change 2`] = `
<article
  class="dusk-balance"
>
  <header
    class="dusk-balance__header"
  >
    <h2
      class="sr-only"
    >
      Your Balance:
    </h2>
  </header>
   
  <p
    class="dusk-balance__dusk"
  >
    <strong>
      4.000.000,000000000
    </strong>
     
    <strong>
      DUSK
    </strong>
  </p>
   
  <p
    class="dusk-balance__fiat dusk-balance__fiat--visible"
  >
    <strong>
      80.000.000,00 €
    </strong>
  </p>
   
</article>
`;
